go_final_project-main/internal/handlers/task.go

```diff
@@ -0,0 +1,213 @@
+			}
+			if givenID > maxID {
+				err = errors.New("given ID is more than number of rows in DB")
+				utils.SendErr(w, err, http.StatusBadRequest)
+				return
+			}
+		}
+		_, err = c.DB.Exec(`UPDATE scheduler SET date = ?, title = ?, comment = ?, repeat = ? WHERE id = ?`,
+			task.Date, task.Title, task.Comment, task.Repeat, task.ID)
+		if err != nil {
+			utils.SendErr(w, err, http.StatusInternalServerError)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		fmt.Fprint(w, "{}")
+	case http.MethodPost:
+		var id struct {
+			ID int64 `json:"id"`
+		}
+		var request models.Task
+		err := json.NewDecoder(r.Body).Decode(&request)
+		if err != nil {
+			utils.SendErr(w, err, http.StatusBadRequest)
+			return
+		}
+
+		err = utils.CheckRequest(request)
+		if err != nil {
+			utils.SendErr(w, err, http.StatusBadRequest)
+			return
+		}
+
+		nextDate, err := utils.CompleteRequest(request)
+		if err != nil {
+			utils.SendErr(w, err, http.StatusBadRequest)
+			return
+		}
+		if request.Date != "" {
+			tmpDate, err := time.Parse("20060102", request.Date)
+			if err != nil {
+				utils.SendErr(w, err, http.StatusBadRequest)
+				return
+			}
+			if request.Date == time.Now().Format("20060102") || time.Now().Before(tmpDate) {
+				nextDate = request.Date
+			}
+		}
+
+		res, err := c.DB.Exec(`INSERT INTO scheduler (date, title, comment, repeat) VALUES (?, ?, ?, ?)`,
+			nextDate, request.Title, request.Comment, request.Repeat)
+		if err != nil {
+			utils.SendErr(w, err, http.StatusInternalServerError)
+			return
+		}
+
+		lastInsertID, err := res.LastInsertId()
+		if err != nil {
+			utils.SendErr(w, err, http.StatusInternalServerError)
+			return
+		}
+		id.ID = lastInsertID
+		response, err := json.Marshal(id)
+		if err != nil {
+			utils.SendErr(w, err, http.StatusInternalServerError)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		w.Write(response)
+	case http.MethodDelete:
+		id := r.FormValue("id")
+		if id == "" {
+			utils.SendErr(w, errors.New("id is empty"), http.StatusBadRequest)
+			return
+		}
+		var maxID int
+		row := c.DB.QueryRow(`SELECT MAX(id) FROM scheduler`)
+		row.Scan(&maxID)
+		if err := row.Err(); err != nil {
+			utils.SendErr(w, err, http.StatusInternalServerError)
+			return
+		}
+		givenID, err := strconv.Atoi(id)
+		if err != nil {
+			err = errors.New("can not parse ID")
+			utils.SendErr(w, err, http.StatusInternalServerError)
+			return
+		}
+		if givenID > maxID {
+			err = errors.New("given ID is more than number of rows in DB")
+			utils.SendErr(w, err, http.StatusBadRequest)
+			return
+		}
+		_, err = c.DB.Exec(`DELETE FROM scheduler WHERE id = ?`, id)
+		if err != nil {
+			utils.SendErr(w, err, http.StatusInternalServerError)
+		}
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		fmt.Fprint(w, "{}")
+	default:
+		err := fmt.Errorf("no request method")
+		utils.SendErr(w, err, http.StatusInternalServerError)
```

Comment:
Классическая ошибка для таких случаев - 405
```suggestion
		utils.SendErr(w, err, http.StatusMethodNotAllowed)
```